app/controllers/oauth/linkedin_controller.rb

```diff
@@ -0,0 +1,72 @@
+class Oauth::LinkedinController < ApplicationController
```

Comment:
This controller is huge. Do you think is there a sense to introduce more top level service objects here?